internal/httphandler/data_request_handler_test.go

```diff
@@ -240,6 +239,265 @@ func TestDataRequestHandler_DataQuery(t *testing.T) {
 	}
 }
 
+func TestDataRequestHandler_DataRangeQuery(t *testing.T) {
+	dbName := "test_database"
+
+	submittingUserName := "alice"
+	cryptoDir := testutils.GenerateTestCrypto(t, []string{"alice", "bob"})
+	aliceCert, aliceSigner := testutils.LoadTestCrypto(t, cryptoDir, "alice")
+	_, bobSigner := testutils.LoadTestCrypto(t, cryptoDir, "bob")
+
+	sigFoo := testutils.SignatureFromQuery(t, aliceSigner, &types.GetDataRangeQuery{
+		UserId:   submittingUserName,
+		DbName:   dbName,
+		StartKey: "key1",
+		EndKey:   "key10",
+		Limit:    10,
+	})
+
+	sigFooNoLimits := testutils.SignatureFromQuery(t, aliceSigner, &types.GetDataRangeQuery{
+		UserId:   submittingUserName,
+		DbName:   dbName,
+		StartKey: "key1",
+		EndKey:   "key10",
+		Limit:    0,
+	})
+
+	testCases := []struct {
+		name               string
+		requestFactory     func() (*http.Request, error)
+		dbMockFactory      func(response *types.GetDataRangeResponseEnvelope) bcdb.DB
+		expectedResponse   *types.GetDataRangeResponseEnvelope
+		expectedStatusCode int
+		expectedErr        string
+	}{
+		{
+			name: "valid get data range with a limit",
```

Comment:
What is the symbol for the "just before first"? is it an empty string?
What is the symbol for the "just after last"? is it an empty string?

I am missing an open range queries like 
start="" / end="key10"
start="key5" / end=""
etc.

Reply:
Done